### ISSUE TITLE
Position not reported when state of covers changes by hard mounted switch.

### ISSUE THREAD

**Issue by DjBac** [iMicknl/ha-tahoma#739]
### Did you read the instructions?

- [X] I have read the [README](https://github.com/iMicknl/ha-tahoma/blob/master/README.md), including the [advanced debugging](https://github.com/iMicknl/ha-tahoma/blob/master/README.md#advanced) section.

### The request

Is there a possibility to be able to show the change of position when we close/open the covers from the hard-mounted switch on the wall?

### Which gateway / hub do you use?

Somfy Tahome

**Comment by tetienne** [comment]
Hi, you didn't provide enough information. Please enable the debug logging as explained [here](https://github.com/iMicknl/ha-tahoma/blob/master/README.md#enable-debug-logging)
And copy paste any log related to this component. Especially the ones starting with `Device added....`

**Comment by iMicknl** [comment]
> When I move a cover from my wall switch, neither the tahoma web interface nor overkiz get updated. Is there an option for setting an interval to get the position of the covers?

Not sure if I understand you correctly, but if tahoma web interface does not support this, we cannot support this either.

**Comment by mstuij** [comment]
I have also asked this a time ago. According to me there was some service like a refresh command.
That triggers the box to get all states of the covers.

@iMicknl the problem was that it was not easy to determine when the update should be requested.

There was some ticket (which I can’t find anymore) about a coordinator that should maybe do the trick. When you open the official Somfy Tahoma app everything is updated by the real state.

But it should be very nice if this is possible in some feature release. I have also everywhere in my house wall mount switches.

I use all options combined: 

- manual
- By the home assistant app
- Automations

the problem is that in your automations you can’t rely on the covers states because someone can control it by the wall switch. So for example:

in my sun protection automations (meassures lux and close/open the covers) i send push notifications to my phone when the covers is opened or closed. 

You have days that the sun shines then some clouds, etc…

So imagine the automation has decided that the cover should closed then you ask the state of the cover. When it is already closed you don’t have to close it another time and you don’t have to send another push notification. 

Now i will always send another close command and push notification.

so for this kind of automations it should be very nice to have real state updates.

**Comment by DjBac** [comment]
> I have also asked this a time ago. According to me there was some service like a refresh command. That triggers the box to get all states of the covers.
> 
> @iMicknl the problem was that it was not easy to determine when the update should be requested.
> 
> There was some ticket (which I can’t find anymore) about a coordinator that should maybe do the trick. When you open the official Somfy Tahoma app everything is updated by the real state.
> 
> But it should be very nice if this is possible in some feature release. I have also everywhere in my house wall mount switches.
> 
> I use all options combined:
> 
> * manual
> * By the home assistant app
> * Automations
> 
> the problem is that in your automations you can’t rely on the covers states because someone can control it by the wall switch. So for example:
> 
> in my sun protection automations (meassures lux and close/open the covers) i send push notifications to my phone when the covers is opened or closed.
> 
> You have days that the sun shines then some clouds, etc…
> 
> So imagine the automation has decided that the cover should closed then you ask the state of the cover. When it is already closed you don’t have to close it another time and you don’t have to send another push notification.
> 
> Now i will always send another close command and push notification.
> 
> so for this kind of automations it should be very nice to have real state updates.

The only option I see now is to use Shelly's i3s or an esp device in combination with the wall switches. This means that the wall switches will control the covers through HA and Tahoma. The only drawback is that the covers will not work if there is no Internet connectivity.

**Comment by mstuij** [comment]
I have everywhere the Somfy Smoove io units on the wall. So the Shelly's and ESP devices are no option.

And the Somfy cloud is very unreliable. So you want to have units that are directly connected to the Soomfy devices.

**Comment by tetienne** [comment]
Closing this issue, as we behave like the tahomalink website.

**Comment by mstuij** [comment]
Some extra information:

When you change the position of a cover with a wall mounted switch like the Somfy Smoove IO:

- you start the Somfy Tahoma App --> the new position is **directly** available in the app
- you start the tahomalink website --> the new position is **directly** available in the website

So they execute the refresh command of the box that requested the states of the IO devices.

So it should be very nice if this command (service) is also available for us or maybe the integration could execute it every 30 seconds?

**Comment by tetienne** [comment]
@mstuij For the cover, it’s not a matter of refresh command. When you login, the 1st step is to call /setup which returns all the states for all the devices, and others data. The fact this endpoint must be called only once per day (it’s an Overkiz requirement).
It’s why we fetch the new events every 30 seconds.

**Comment by iMicknl** [comment]
@tetienne this is not fully true. TaHoma also calls the refresh states endpoint on login (see my very old PR), which will update the state for these devices.

**Comment by jakubskopek** [comment]
Hi, I have similar problem with my garage gate. I have Somfy Connexoon IO hub unit. If I open/close gate via home assistant, the status is updated. Even if I use hardware switches, the status is updated. But I have a gate setting, that the gate closes automatically 10 seconds after it was opened (if there is an obstacle, the time is longer). And after this, the status remains as "opened" for exactly one hour (is this the update interval?) after closing. I updated a home assistant core and even after restart, the gate had still "opened" status. It happend with both core and HACS integration. In the Somfy mobile app (Connexoon Access), the status is shown correctly when I click "check status".
Is there any possibility to use some service where can I ask the current state in an automation? Or to change update interval when the gate is opened and reset it after it is closed?
